test/test_ss_parsimony_restraint2.py

```diff
@@ -0,0 +1,131 @@
+        IMP.atom.SecondaryStructureResidue.setup_particle(pi, 0, 1, 0)
+    elif element[-1] == 'C':
+        IMP.atom.SecondaryStructureResidue.setup_particle(pi, 0, 0, 1)
+    else:
+        IMP.atom.SecondaryStructureResidue.setup_particle(pi, 0.33, 0.33, 0.33)
+        print("secondary structure type is not defined for the corresponding element")
+
+    se = IMP.threading.StructureElement(pi)
+    se.set_keys_are_optimized(True)
+
+    return se
+
+
+# The "true" sequence
+#  MET at 11 and 37
+seq = "SQPAKKTYTWNTKEEAKQAFKEALKEKRVPSNASWEQAMKMIINDPRYSALAKLSEKKQAFNAYKVQTEK"
+# A toy system consisting of the two helices and 50 residues.
+
+# Restraint Weights
+length_slope = 0.1
+xl_slope = 0.1
+semet_slope = 0.1
+psipred_slope = 0.1
+
+
+m = IMP.Model()
+######################################
+DATADIR = os.path.abspath(os.path.join(os.path.dirname(__file__), 'input'))
+pdbfile = os.path.join(DATADIR, "pdb2lv9_A.ent")
+root_hier = IMP.atom.read_pdb(pdbfile, m)
+#IMP.atom.show_molecular_hierarchy(root_hier)
+
+# Define Structural Elements
+# These are the residues in the PDB that correspond to structural elements.
+# (start_res, length, SSID)
+#-------------------------
+#elements=[(3,8,'H'),(19,19,'H')]#, (56,8,'H')]
+elements=[(15,11,'H'),(13,16,'C'), (48,12,'H')]
+
+se = []
+
+for e in elements:
+    se.append(setup_structural_element(root_hier, e))
+
+se = sort_ses(se)
+
+#######################
+# Set up Sequence
+#######################
+seq_chain = IMP.atom.Chain.setup_particle(IMP.Particle(m), "A")
+seq_chain.set_name(seq_chain.get_id())
+root_hier.add_child(seq_chain)
+for i in range(len(seq)):
+    res = IMP.atom.Residue.setup_particle(IMP.Particle(m),
+                                                    IMP.pmi.alphabets.amino_acid.get_residue_type_from_one_letter_code(seq[i]),
+                                                    i+1)
+    IMP.atom.Mass.setup_particle(res.get_particle(), IMP.atom.get_mass(res.get_residue_type()))
+    IMP.core.XYZR.setup_particle(res.get_particle())
+    IMP.atom.SecondaryStructureResidue.setup_particle(res.get_particle(), 0.8, 0, 0.2)
+    seq_chain.add_child(res)
+
+#######################
+# Set up Scoring Function
+#######################
+
+# SS Parsimony Restraint
+ssp_weight = 1.0
+
+# Restraint evaluated at the SE level? Pass SE and seq_chain.  
+# Have restraint look up residue from the seq_chain and get SS
+
+a = IMP.atom.Hierarchy(seq_chain.get_particle()).get_children()
+se_id = [s.get_particle_index() for s in se]
+print(len(se_id))
+print("get first sequence residue's all probabilities", IMP.atom.SecondaryStructureResidue(m, a[0].get_particle_index()).get_all_probabilities())
+print("get first SE residue's all probabilities", IMP.atom.SecondaryStructureResidue(m, se_id[0]).get_all_probabilities())
+
+par_res = IMP.threading.SecondaryStructureParsimonyRestraint(m, se_id, seq_chain.get_particle_index(), 1.0)
+a_value  = par_res.unprotected_evaluate(None)
+print(a_value)
+
+
+
```

Comment:
This test doesn't appear to actually, well, test anything (other than "hope it doesn't crash"). (I give you a pass again here because the original tests were like this too.) Better to use `unittest` and then `assert` on a bunch of the outputs. (Or if you hate `unittest` you could use `pytest` tests instead, since the CI does use pytest.) Printing stuff is no use at all because nobody ever looks at the test output. See most IMP/PMI tests for examples.